pkg/controllers/metrics/node/controller.go

```diff
@@ -0,0 +1,88 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-core/pkg/controllers/state"
+	"github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/utils/resources"
+)
+
+const pollingPeriod = 5 * time.Second
+
+type Controller struct {
+	cluster *state.Cluster
+}
+
+func NewController(cluster *state.Cluster) *Controller {
+	return &Controller{
+		cluster: cluster,
+	}
+}
+
+func (c *Controller) Name() string {
+	return "metric_scraper"
+}
+
+func (c *Controller) Builder(_ context.Context, mgr manager.Manager) controller.Builder {
+	return controller.NewSingletonManagedBy(mgr)
+}
+
+func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	// Reset all the gauges ahead of setting the new gauges with the new labels
+	for _, v := range gauges {
+		v.Reset()
+	}
```

Comment:
What happens if a scraper reads the gauge right here?

Reply:
Will test out if there's a race here and what the availability promises look like